osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSpinner.cs

```diff
@@ -173,7 +178,14 @@ protected override void UpdateHitStateTransforms(ArmedState state)
         {
             base.UpdateHitStateTransforms(state);
 
-            this.FadeOut(160).Expire();
+            this.FadeOut(fade_out_duration).OnComplete(_ =>
+            {
+                // looping sample should be stopped here as it is safer than running in the OnComplete
+                // of the volume transition above.
+                spinningSample.Stop();
+            });
```

Comment:
`UpdateHitStateTransforms()` also handles `ArmedState.Idle`. This logic should probably not execute in that state.

Reply:
I don't think it can cause any harm by doing so, but I guess it's safe to not run.

Reply:
Actually I'm not sure about this any more. It was previously here and is likely being used to actually decide the lifetime (the `FadeOut` before the `Expire`). I think it's probably fine to leave as is?